### PR TITLE
Modify testrbd.py to support diffrent branch checkout

### PR DESCRIPTION
**Pull request by ckulal**
Modify testrbd.py to clone based on branch given

logs:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1626080525853/
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1626083263551/

Signed-off-by: ckulal <ckulal@redhat.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs

# Checklist:

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

### CODE REVIEW

File: tests/rbd/test_rbd.py [hunk 1]
```diff
@@ -31,7 +31,13 @@ def one_time_setup(node, rhbuild, branch: str) -> None:
     """
     try:
         node.exec_command(cmd="ls ceph")
+        out, err = node.exec_command(cmd="cd ceph; git branch")
```
Comment (psathyan) [inline tests/rbd/test_rbd.py:34]: Instead of `git branch`, could you use `git rev-parse --abbrev-ref HEAD`? It would return only the branch name

**Comment by rakeshgm** [comment]
The commit looks good, its nice to check on which branch we are on before doing a checkout of the specific branch. But i think we can definitely look for improvement here. 
If we clone ceph repo, the size is around 1 GB and cephci takes a long time to clone. so to reduce the size, time and instead of pulling all the branches, we can just do a clone of that branch and just take the recent commits from that branch using dept =1 

example:

```
git clone --branch <branch_name> --single-branch --depth 1 <repo>
```
you can even checkout only a specific folder too and not download all the src files which are not needed.

@ckulal can you take a look at it.

File: tests/rbd/test_rbd.py [hunk 2]
```diff
@@ -31,8 +31,7 @@ def one_time_setup(node, rhbuild, branch: str) -> None:
     """
     try:
         node.exec_command(cmd="ls ceph")
-
-        return
+        node.exec_command(cmd="rm -rf ceph")
     except BaseException:  # noqa
         pass
```
Comment (psathyan) [inline tests/rbd/test_rbd.py:37]: The try except block is not required as the plan is to remove the folder if found. Hence, we could avoid the block and have `node.exec_command(cmd="rm -rf ceph")`

**Review by psathyan — COMMENTED** [review]
This look good... a minor improvement.. Another option is to add it to the list of `commands`

**Review by psathyan — APPROVED** [review]
LGTM

**Pull request merged**